src/webgpu/shader/execution/robust_access.spec.ts

```diff
@@ -310,7 +310,13 @@ g.test('linear_memory')
           switch (access) {
             case 'read':
               {
-                const exprLoadElement = isAtomic ? `atomicLoad(&${exprElement})` : exprElement;
+                let exprLoadElement = isAtomic ? `atomicLoad(&${exprElement})` : exprElement;
+                if (storageClass === 'uniform' && containerType === 'array') {
+                  // Scalar types will be wrapped in a vec4 to satisfy array element size
+                  // requirements for the uniform address space, so we need an additional index
+                  // accessor expression.
+                  exprLoadElement += '[0]';
```

Comment:
This is a bit gross. I'd welcome suggestions for better ways to handle this, since anyone else that calls `generateTypes()` would need to do the same.